src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CreateCaseCallbackService.java

```diff
@@ -163,32 +167,57 @@ private ProcessResult tryCreateNewCase(
             return new ProcessResult(emptyList(), singletonList(AWAITING_PAYMENTS_MESSAGE));
         } else {
             List<Long> ids = ccdApi.getCaseRefsByBulkScanCaseReference(exceptionRecord.id, configItem.getService());
-            if (ids.isEmpty()) {
-                return ccdNewCaseCreator.createNewCase(
-                    exceptionRecord,
-                    configItem,
-                    ignoreWarnings,
-                    idamToken,
-                    userId,
-                    exceptionRecordData
-                );
-            } else if (ids.size() == 1) {
-                final Map<String, Object> finalizedExceptionRecordData =
-                    exceptionRecordFinalizer.finalizeExceptionRecord(
-                        exceptionRecordData.getData(),
-                        ids.get(0),
-                        CcdCallbackType.CASE_CREATION
-                    );
-                return new ProcessResult(finalizedExceptionRecordData);
-            } else {
+            if (ids.size() > 1) {
                 String msg = String.format(
                     "Multiple cases (%s) found for the given bulk scan case reference: %s",
                     ids.stream().map(String::valueOf).collect(joining(", ")),
                     exceptionRecord.id
                 );
                 log.error(msg);
                 throw new MultipleCasesFoundException(msg);
+            } else {
+                Either<ProcessResult, Long> result = ids
+                    .stream()
+                    .findFirst()
+                    .map(Either::<ProcessResult, Long>right)
+                    .orElseGet(() -> ccdNewCaseCreator.createNewCase(
+                        exceptionRecord,
+                        configItem,
+                        ignoreWarnings,
+                        idamToken,
+                        userId
+                    ));
+
+                Either<ProcessResult, ProcessResult> processedResult = result
+                    .map(id -> {
+                        tryPublishPaymentMessage(configItem.getService(), exceptionRecordData, id);
+
+                        return new ProcessResult(
+                            exceptionRecordFinalizer.finalizeExceptionRecord(
+                                exceptionRecordData.getData(),
+                                id,
+                                CcdCallbackType.CASE_CREATION
+                            )
+                        );
+                    });
+
+                return processedResult.isLeft() ? processedResult.getLeft() : processedResult.get();
```

Comment:
The logic here is quite simple, but the code doesn't read very easily. There are transformations on `Either<ProcessResult, ProcessResult>`, where the left `ProcessResult` represents a different situation than the right one and at some point the right one gets transformed into a different `ProcessResult`. Worth making this code more explicit.